### DISCUSSION TITLE
Dynamic cache keys for #[Computed] similiar to #[On]

### DISCUSSION THREAD

**Discussion by LarsWiegers** [livewire/livewire#8532] [3. Feature ideas]
As defined here: https://livewire.laravel.com/docs/events#listening-for-dynamic-event-names. Event listeners using the #[On] php attribute can use dynamic listeners based on properties defined on the livewire component.

It would be really cool to do the same for the cache key of the #[Computed] property. 

Why i want this:
We have an component that contains a datatable. 

Loading these rows takes a while. 
We currently define our own caching strategy to reduce the load time for the user. 
This caching is defined based on the id + filters + what paginated page they are on. 

If we could use a property here that we update whenever the paginated  page or filter is changed we could use the livewire caching and remove our custom caching solution. 

Example code:

```
// Dynamically updated whenever filters have changed
public string $cacheKey = "";

#[Computed(persist: true, key: 'rows_{cacheKey}')]
public function rows() {
   return $this->applyFilters(Item::query());
}
```

**Comment by djfrailey** [comment]
I agree that this would add to the developer experience of Livewire, however while I was working on a proof-of-concept for this I realized that what you want to do could potentially be accomplished through the use of tags.

See: https://github.com/livewire/livewire/blob/main/src/Features/SupportComputed/BaseComputed.php#L90

I haven't tried it yet myself but plan to in the very near future.

**Reply by francoism90** [reply]
@djfrailey You will have the same problem, it's not possible to inject an (model) ID into the cache tags.

**Comment by francoism90** [comment]
Created a PR: https://github.com/livewire/livewire/pull/8784

It would be useful, since you could create multiple computed methods, that don't have to be handled by model logic/actions, but still can be flushed when needed to. :)
